recipes/mender-client/debian-3.5.x/preinst

```diff
@@ -0,0 +1,64 @@
+
+set -e
+
+ACTION="$1"
+OLD_VERSION="$2"
+NEW_VERSION="$3"
+
+case "$ACTION" in
+    upgrade)
+        case "$OLD_VERSION" in
+            4.*)
+                # Downgrading from v4 we need to take back ownership of the conffiles that were
+                # split into mender-auth and mender-update packages and disable the systemd services
+                # mender-auth files
+                dpkg-divert --add --no-rename /usr/share/mender/identity/mender-device-identity
+                dpkg-divert --add --no-rename /etc/mender/identity/mender-device-identity
+                dpkg-divert --add --no-rename /usr/share/dbus-1/system.d/io.mender.AuthenticationManager.conf
+                # mender-update files
+                dpkg-divert --add --no-rename /etc/mender/inventory/mender-inventory-bootloader-integration
+                dpkg-divert --add --no-rename /etc/mender/inventory/mender-inventory-hostinfo
+                dpkg-divert --add --no-rename /etc/mender/inventory/mender-inventory-network
+                dpkg-divert --add --no-rename /etc/mender/inventory/mender-inventory-os
+                dpkg-divert --add --no-rename /etc/mender/inventory/mender-inventory-provides
+                dpkg-divert --add --no-rename /etc/mender/inventory/mender-inventory-rootfs-type
+                dpkg-divert --add --no-rename /etc/mender/inventory/mender-inventory-update-modules
+                dpkg-divert --add --no-rename /usr/share/mender/modules/v3/deb
+                dpkg-divert --add --no-rename /usr/share/mender/modules/v3/directory
+                dpkg-divert --add --no-rename /usr/share/mender/modules/v3/docker
+                dpkg-divert --add --no-rename /usr/share/mender/modules/v3/rootfs-image
+                dpkg-divert --add --no-rename /usr/share/mender/modules/v3/rpm
+                dpkg-divert --add --no-rename /usr/share/mender/modules/v3/script
+                dpkg-divert --add --no-rename /usr/share/mender/modules/v3/single-file
+                dpkg-divert --add --no-rename /usr/share/mender/inventory/mender-inventory-bootloader-integration
+                dpkg-divert --add --no-rename /usr/share/mender/inventory/mender-inventory-hostinfo
+                dpkg-divert --add --no-rename /usr/share/mender/inventory/mender-inventory-network
+                dpkg-divert --add --no-rename /usr/share/mender/inventory/mender-inventory-os
+                dpkg-divert --add --no-rename /usr/share/mender/inventory/mender-inventory-provides
+                dpkg-divert --add --no-rename /usr/share/mender/inventory/mender-inventory-rootfs-type
+                dpkg-divert --add --no-rename /usr/share/mender/inventory/mender-inventory-update-modules
+                dpkg-divert --add --no-rename /etc/mender/scripts/version
+                # stop and disable the v4 services
+                if systemctl is-active mender-authd; then
+                    systemctl stop mender-authd
+                    systemctl disable mender-authd
```

Comment:
I think we need to remove these packages, we can't just disable them. Otherwise the system will be left with broken zombie packages afterwards, that have some, but not all of the files.

In fact, I'm wondering if removing should be done *instead of* using dpkg-divert, because the ordering is opposite here, so the files should already be gone by the time mender-client 3.5.x tries to put them in place.